pkg/repositories/models/node_execution.go

```diff
@@ -29,9 +29,16 @@ type NodeExecution struct {
 	NodeExecutionUpdatedAt *time.Time
 	Duration               time.Duration
 	NodeExecutionEvents    []NodeExecutionEvent
+	// Metadata about the node execution.
+	NodeExecutionMetadata []byte
+	// Parent that spawned this node execution - value is empty for executions at level 0
+	ParentID *uint `sql:"default:null"`
+	// List of child node executions - for cases like Dynamic task, sub workflow, etc
+	ChildNodeExecutions []NodeExecution `gorm:"foreignkey:ParentID"`
```

Comment:
because i'm dense, can you maybe explain to me what this means?  usually in a has-many relationship, it means there are multiple rows of the thing that you have many of.  in this case however, we're storing things as a list?  or does it just mean that multiple children rows can have the same ParentID?

I think it'd be super helpful to see the db table before and after this change.  Maybe I'll play around when I have some time.

Reply:
Multiple children rows can have the same ParentID

```         Column           |           Type           | Collation | Nullable |                   Default                   
---------------------------+--------------------------+-----------+----------+---------------------------------------------
 id                        | integer                  |           | not null | nextval('node_executions_id_seq'::regclass)
 created_at                | timestamp with time zone |           |          | 
 updated_at                | timestamp with time zone |           |          | 
 deleted_at                | timestamp with time zone |           |          | 
 execution_project         | text                     |           | not null | 
 execution_domain          | text                     |           | not null | 
 execution_name            | text                     |           | not null | 
 node_id                   | text                     |           | not null | 
 phase                     | text                     |           |          | 
 input_uri                 | text                     |           |          | 
 closure                   | bytea                    |           |          | 
 started_at                | timestamp with time zone |           |          | 
 node_execution_created_at | timestamp with time zone |           |          | 
 node_execution_updated_at | timestamp with time zone |           |          | 
 duration                  | bigint                   |           |          | 
 parent_task_execution_id  | integer                  |           |          | 
 node_execution_metadata   | bytea                    |           |          | 
 parent_id                 | integer                  |           |          | 
 error_kind                | text                     |           |          | 
 error_code                | text                     |           |          | 
 cache_status              | text                     |           |          | 
Indexes:
    "node_executions_pkey" PRIMARY KEY, btree (execution_project, execution_domain, execution_name, node_id)
    "idx_node_executions_deleted_at" btree (deleted_at)
    "idx_node_executions_error_kind" btree (error_kind)
    "idx_node_executions_id" btree (id)
    "idx_node_executions_node_id" btree (node_id)
    "idx_node_executions_parent_task_execution_id" btree (parent_task_execution_id)

```